README.md

```diff
@@ -1,2 +1,175 @@
-# go-libp2p-relay
-Circuit Switching for libp2p
+# go-libp2p-circuit-relay
+> Circuit Switching in libp2p
+
+Table of Contents
+- [Overview](#overview)
+- [Dramatization](#dramatization)
+- [Addressing](#addressing)
+- [/ipfs/relay/circuit stream protocol](#ipfsrelaycircuit-stream-protocol)
+- [Interfaces](#interfaces)
+- [Removing existing relay protocol](#removing-existing-relay-protocol)
+
+
+### Overview
+
+The circuit relay is a means of establishing connectivity between
+libp2p nodes (such as IPFS) that wouldn't otherwise be able to connect to each other.
+
+This helps in situations where nodes are behind NAT or reverse proxies,
+or simply don't support the same transports (e.g. go-ipfs vs. browser-ipfs).
+libp2p already has modules for NAT ([go-libp2p-nat][go-libp2p-nat]),
+but these don't always do the job, just because NAT traversal is a clusterfuck.
```

Comment:
yes, did this in the spirit of banana-driven writing. if you can't come up with a good term, insert one which is so bad you *have* to replace it :)